dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/hibernate/HibernateUserStore.java

```diff
@@ -245,7 +245,7 @@ private Query<?> getUserQuery( UserQueryParams params, List<String> orders, bool
             {
                 hql += hlp.whereAnd() + " (";
 
-                for ( OrganisationUnit ou : params.getDataViewOrganisationUnits() )
+                for ( OrganisationUnit ou : params.getOrganisationUnits() )
```

Comment:
This is better. Sorry for harping on this. I think it is actually possible to do away with 2 of the 3 if code blocks for org unit filtering entirely. The only piece that is now different across the 3 blocks is the join field alias. I.e. we have:

`left join u.organisationUnits ou `
`left join u.dataViewOrganisationUnits dwou`
`left join u.teiSearchOrganisationUnits tsou`
 
 This can be parameterized to:
 
 `left join u.${org_unit_field} ou`
 
 and `${org_unit_field}` can be inferred from the user org unit type.
 
 The "query parameters" part further down in the method must be updated too. 
 
 That way we collapse 3 blocks of if/for into 1 it could seem.